infra/main.bicep

```diff
@@ -56,22 +56,27 @@ module appEnv './app/app-env.bicep' = {
     containerRegistryName: !empty(containerRegistryName) ? containerRegistryName : '${abbrs.containerRegistryRegistries}${resourceToken}'
     location: location
     logAnalyticsWorkspaceName: monitoring.outputs.logAnalyticsWorkspaceName
+    applicationInsightsName: monitoring.outputs.applicationInsightsName
+    daprEnabled: true
   }
 }
 
-// Worker 
+// Worker
 module worker './app/worker.bicep' = {
   name: workerServiceName
   scope: rg
   params: {
     name: !empty(workerContainerAppName) ? workerContainerAppName : '${abbrs.appContainerApps}${workerServiceName}-${resourceToken}'
     location: location
     imageName: workerImageName
-    applicationInsightsName: monitoring.outputs.applicationInsightsName
     containerAppsEnvironmentName: appEnv.outputs.environmentName
     containerRegistryName: appEnv.outputs.registryName
     serviceName: workerServiceName
+    managedIdentityName: security.outputs.managedIdentityName
   }
+  dependsOn: [
```

Comment:
I think with the proper use of outputs, you can avoid having to use "dependsOn" in most scenarios.